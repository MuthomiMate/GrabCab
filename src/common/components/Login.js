import React, { Component } from "react";
import { View, TextInput, StyleSheet, Button, TouchableOpacity, Text } from "react-native"

class Login extends Component {
    constructor(props){
        super(props)
        this.state = {
            email: '',
            password: ''
        }
    }

    render() {
        const { style } = this.props;
        return (
            <View
            style={style}
            >
                <View style={styles.inputs}>
                    <View style={styles.email}>
                        <TextInput
                            style={{textAlign: "center", height: "100%"}}
                            placeholder="Email"
                            keyboardType="email-address"
                        />
                    </View>
                    <View style={styles.password}>
                        <TextInput
                            placeholder="Password"
                            style={{textAlign: "center", height: "100%"}}
                        />
                    </View>
                </View>
                <View style={styles.buttons}>
                    <Button
                        title="Register"
                        color="#FFFFFF"
                    />
                    <TouchableOpacity style={styles.touchable}>
                        <Text style={{color: "#FFFFFF"}}> Forgot Password?</Text>
                    </TouchableOpacity>
                    <Button
                        title="Login"
                        color="#FFFFFF"
                    />
                </View>
            </View>
        )
    }
}
const styles = StyleSheet.create({
    inputs: {
        flex: 1,
        width: "100%",
        height: "80%",
        borderRadius: 10,
        flexDirection: "column",
        backgroundColor: "#FFFFFF"
    },
    email: {
        borderBottomColor: "black",
        borderBottomWidth: 1,
        height: "50%"
    },
    password: {
        borderTopColor: "black",
        borderTopWidth: 1,
        height: "50%"
    },
    buttons: {
        flex: 1,
        fontSize: 8,
        width: "100%",
        height:"20%",
        flexDirection: "row"
    },
    touchable: {
        marginTop: "5%",

    }

})
export default Login