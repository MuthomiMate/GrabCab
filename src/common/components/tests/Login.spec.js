import React from "react";
import { shallow } from "enzyme";
import Login from "../Login";

describe("<Login />", () => {
    const tesProps = {
        style: {}
    }

    it("renders successfully", () => {
        const wrapper = shallow(<Login {...tesProps}/>)
        expect(wrapper).toMatchSnapshot()
    });
})