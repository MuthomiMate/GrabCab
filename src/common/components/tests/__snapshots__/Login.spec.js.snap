// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> renders successfully 1`] = `
<View
  style={Object {}}
>
  <View
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "borderRadius": 10,
        "flex": 1,
        "flexDirection": "column",
        "height": "80%",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "borderBottomColor": "black",
          "borderBottomWidth": 1,
          "height": "50%",
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        keyboardType="email-address"
        placeholder="Email"
        rejectResponderTermination={true}
        style={
          Object {
            "height": "100%",
            "textAlign": "center",
          }
        }
        underlineColorAndroid="transparent"
      />
    </View>
    <View
      style={
        Object {
          "borderTopColor": "black",
          "borderTopWidth": 1,
          "height": "50%",
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        placeholder="Password"
        rejectResponderTermination={true}
        style={
          Object {
            "height": "100%",
            "textAlign": "center",
          }
        }
        underlineColorAndroid="transparent"
      />
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "fontSize": 8,
        "height": "20%",
        "width": "100%",
      }
    }
  >
    <Button
      color="#FFFFFF"
      title="Register"
    />
    <TouchableOpacity
      activeOpacity={0.2}
      style={
        Object {
          "marginTop": "5%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
          }
        }
      >
         Forgot Password?
      </Text>
    </TouchableOpacity>
    <Button
      color="#FFFFFF"
      title="Login"
    />
  </View>
</View>
`;
